app/commands/git/pull_request/sync_repo.rb

```diff
@@ -0,0 +1,146 @@
+module Git
+  class PullRequest
+    class SyncRepo
+      include Mandate
+
+      initialize_with :repo
+
+      def call
+        pull_requests.each do |pr|
+          Git::PullRequest::CreateOrUpdate.(pr[:pr_id],
+            pr_number: pr[:pr_number],
+            author: pr[:author],
+            merged_by: pr[:merged_by],
+            repo: pr[:repo],
+            reviews: pr[:reviews],
+            data: pr)
+        end
+      end
+
+      private
+      memoize
+      def pull_requests
+        cursor = nil
+        results = []
+
+        loop do
+          page_data = fetch_page(cursor)
+
+          # TODO: filter out PRs we want to ignore (e.g. the v3 bulk rename PRs)
+          results += pull_requests_from_page_data(page_data)
+
+          break results unless page_data[:data][:repository][:pullRequests][:pageInfo][:hasNextPage]
```

Comment:
I'd use `dig` here.

Reply:
I like you use of `break results` here.

Reply:
I tried that a while back but I couldn't get it to work. I now know why: `page_data` is not a regular hash but a `Sawyer::Resource`, which is an octokit type. I'm now calling `to_h` on the `Sawyer::Resource` and that makes `dig` work.